src/pip/_internal/commands/cache.py

```diff
@@ -175,7 +175,7 @@ def remove_cache_items(self, options: Values, args: List[Any]) -> None:
             files += self._find_http_files(options)
 
         if not files:
-            raise CommandError("No matching packages")
+            logger.warning("No matching packages")
```

Comment:
We can probably remove this entire conditional.

It's not like this warning is super useful; given that the next line in the output is "Files removed: 0".

Reply:
I actually like this warning, people expect this command to remove wheels, and it seems appropriate to at least add a warning here.

Another point for the warning is that this function is also used by `pip cache remove` which lets people provide a pattern. It's nice to have a warning in case that the pattern is broken.
I can even change this to `No matching packages for pattern {pattern}`

Reply:
I like `No matching packages for pattern {pattern}`, but I see how it could confuse users trying to purge cache with `No matching packages for pattern *`.

How about `logger.warning("No matching packages " + "to purge" if args[0] == "*" else f"for pattern {args[0]}")`? That seems a bit convoluted though...

Reply:
see now @ziebam @pradyunsg